### DISCUSSION TITLE
What's the fastest model on anomalib?

### DISCUSSION THREAD

**Discussion by cory0619** [open-edge-platform/anomalib#1739] [General]
I've checked the inference time with PatchCore and EfficientAd.
With my dataset, EfficientAD is faster and more precise than PatchCore regarding anomaly detection.
Is there any better model than EfficientAD comparing with inference time and accuracy on anomalib?

I think it depends on dataset properties but, I wonder model's overall performance regarding unsupervised training.

Thanks in advance.

**Comment by blaz-r** [comment]
Hello, while we don't have the exact benchmarks for inference speed, EfficientAD is probably one of the fastest models available right now and it also offers great performance. Therefore I think it's a good choice if you manage to achieve good performance on your dataset.

**Reply by cory0619** [reply]
@blaz-r  Thank you!

**Comment by yxteo2** [comment]
> I've checked the inference time with PatchCore and EfficientAd. With my dataset, EfficientAD is faster and more precise than PatchCore regarding anomaly detection. Is there any better model than EfficientAD comparing with inference time and accuracy on anomalib?
> 
> I think it depends on dataset properties but, I wonder model's overall performance regarding unsupervised training.
> 
> Thanks in advance.

Hi can you share your setting on EfficientAd? Mine took forever to run

**Comment by yxteo2** [comment]
> I've checked the inference time with PatchCore and EfficientAd. With my dataset, EfficientAD is faster and more precise than PatchCore regarding anomaly detection. Is there any better model than EfficientAD comparing with inference time and accuracy on anomalib?
> 
> I think it depends on dataset properties but, I wonder model's overall performance regarding unsupervised training.
> 
> Thanks in advance.

I tested FastFlow, EfficientAd, Dfm, and Patchcore
It seems that EfficientAd is the slowest

The time they took to process an image is below:
- Dfm: 25 ms
- FastFlow: 30 ms
- PatchCore: 90-100 ms
- EfficientAd: 150 ms
